### PR TITLE
new atomic pkg

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Removes dependency for plugin/evm package for atomic tx types.

## How this works

This pull request includes several important changes related to the `plugin/evm` package, focusing on renaming files, updating imports, and modifying functions to improve code organization and functionality.

### File Renaming and Package Changes:
* Renamed `plugin/evm/codec.go` to `plugin/evm/atomic/codec.go` and updated the package declaration to `atomic`.
* Renamed `plugin/evm/export_tx.go` to `plugin/evm/atomic/export_tx.go` and updated the package declaration to `atomic`.
* Renamed `plugin/evm/import_tx.go` to `plugin/evm/atomic/import_tx.go` and updated the package declaration to `atomic`.

### Import Updates:
* Added import for `github.com/ava-labs/coreth/plugin/evm/client` in `plugin/evm/admin.go`.
* Updated various import paths to reflect the new `atomic` package and added necessary imports in renamed files. [[1]](diffhunk://#diff-c828570d214549415eeb1934b82a0490f7a02368769d234f5e8451e4b5c1e9bbR16-R23), [[2]](diffhunk://#diff-149f21dc196b9ef59c37af420acd7445323ee5a1da4eec76b37e8fda9a0754b1L13)

### Function Modifications:
* Modified `SetLogLevel` function in `plugin/evm/admin.go` to use `client.SetLogLevelArgs` instead of a locally defined struct.
* Updated error handling and constants in `plugin/evm/atomic/export_tx.go` to use exported error variables. [[1]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL35-R41) [[2]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL76-R99) [[3]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL103-R108) [[4]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL113-R128)
* Refactored the `SemanticVerify` function to use the `Backend` struct instead of `VM`. [[1]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL179-R191) [[2]](diffhunk://#diff-9112fb95910e41b6c5b661fa228c9061a503dd48074da7fc7d885c0f2eb8763cL202-R211)

### New Functions:
* Added `GetSpendableFunds` and `GetSpendableAVAXWithFee` functions to handle the retrieval of spendable funds for transactions.

These changes aim to improve code modularity, maintainability, and clarity by reorganizing the `plugin/evm` package and enhancing function implementations.

## How this was tested

UT should cover this

## Need to be documented?

No

## Need to update RELEASES.md?

No

### CODE REVIEW

File: plugin/evm/client/client.go [hunk 1]
```diff
@@ -229,8 +300,8 @@ func (c *client) SetLogLevel(ctx context.Context, level slog.Level, options ...r
 }
 
 // GetVMConfig returns the current config of the VM
-func (c *client) GetVMConfig(ctx context.Context, options ...rpc.Option) (*Config, error) {
-	res := &ConfigReply{}
-	err := c.adminRequester.SendRequest(ctx, "admin.getVMConfig", struct{}{}, res, options...)
-	return res.Config, err
-}
+// func (c *client) GetVMConfig(ctx context.Context, options ...rpc.Option) (*Config, error) {
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply plugin/evm/client/client.go:303]: fixed in #706

File: plugin/evm/atomic/import_tx.go [hunk 2]
```diff
@@ -275,28 +287,11 @@ func (utx *UnsignedImportTx) AtomicOps() (ids.ID, *atomic.Requests, error) {
...
-	return vm.newImportTxWithUTXOs(chainID, to, baseFee, kc, atomicUTXOs)
-}
-
-// newImportTx returns a new ImportTx
-func (vm *VM) newImportTxWithUTXOs(
+// NewImportTx returns a new ImportTx
+func NewImportTx(
+	ctx *snow.Context,
+	rules params.Rules,
+	clk mockable.Clock,
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply plugin/evm/atomic/import_tx.go]: done

File: utils/utils.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // (c) 2019-2024, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
-package atomic
+package utils
```
Comment (darioush) [inline utils/utils.go:4]: Should we keep this in a separate package like `plugin/evm/client`? IMO we can move `plugin/evm/client.go` to that new package (as we need somewhere to put `NewCChainClient` that doesn't import `plugin/evm` soon)

Comment (ceyonur) [reply utils/utils.go:4]: that `utils` is already in a different pkg, just not in the same one with client. Do you want to move it to the same client pkg? Also the new `client` package does not import `plugin/evm` (just `plugin/evm/atomic`).

Comment (darioush) [reply utils/utils.go:4]: The main concern is that the `utils` package may end up importing more stuff than we want, for example it's already importing avalanchego package, which can cause circular issues again (as the clients are using this function eg in tests)

Comment (ceyonur) [reply utils/utils.go]: It turned out that we don't actually need those methods in there. Exported those from avalanchego

**Review by darioush — APPROVED** [review]

**Pull request merged**